reactor-test/src/main/java/reactor/test/ValueFormatters.java

```diff
@@ -0,0 +1,434 @@
+/*
+ * Copyright (c) 2011-2019 Pivotal Software Inc, All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *        https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package reactor.test;
+
+import java.util.Collection;
+import java.util.Spliterator;
+import java.util.function.BiFunction;
+import java.util.function.Function;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import java.util.stream.StreamSupport;
+
+import reactor.core.Fuseable;
+import reactor.core.publisher.Signal;
+import reactor.util.annotation.Nullable;
+
+/**
+ * An utility class to create {@link ToStringConverter} {@link Function} that convert objects to {@link String}.
+ * They can be applied to any {@link Object} but will restrict the types they actually
+ * convert by filtering and defaulting to {@link String#valueOf(Object)}.
+ * <p>
+ * Also defines {@link Extractor} {@link BiFunction} that extract multiple elements from
+ * containers and applies a {@link Function} (most probably a {@link ToStringConverter})
+ * to its elements, reforming a complete {@link String} representation. Again, this matches
+ * on at least a given {@link Class} and potentially an additional {@link Predicate}.
+ * <p>
+ * An additional static utility method {@link #convertVarArgs(ToStringConverter, Collection, Object...)} can
+ * be used to apply such functions and  to varargs.
+ *
+ * @author Simon Baslé
+ */
+public final class ValueFormatters {
+
+	private ValueFormatters() {}
+
+	/**
+	 * A generic {@link Object} to {@link String} conversion {@link Function} which is
+	 * also a {@link Predicate}, and which only applies a custom conversion to targets that
+	 * match said {@link Predicate}. Other targets are converted using {@link String#valueOf(Object)}.
+	 */
+	interface ToStringConverter extends Predicate<Object>, Function<Object, String> {}
```

Comment:
`ToStringConverter` needs to be `public`

Reply:
ah good catch, both interfaces needs to be `public` yeah